### DISCUSSION TITLE
ESP32-C3 capabilities?

### DISCUSSION THREAD

**Discussion by roguestreak** [marcel-licence/esp32_basic_synth#88] [General]
Hi Marcel
I've just ordered both an ESP32 (wroom32 variant) and a pi pico (RP2040) for experimentation with your synths, potentially seeing if the RP2040 can also receive USB MIDI at the same time, otherwise possibly communicating between the two...
Anyway, curious if you think the new ESP32-C3 would have enough grunt to run the synth, given it's a single core and different architecture?
I do have the DAC, but will also test DAC-less with output straight from the chip

**Comment by marcel-licence** [comment]
Hi,
sounds good. I am currently searching for some DAC-less example running with Arduino on the ESP32 with no success yet. If you want to run only the sound engine MIDI -> Audio, it should work with a single core. It might be impossible to drive I2C or SPI at the same time depending on the library / implementation. 
If you need the library, some of the pre-compiled files, I should compile and upload it. I've prepared it for the ESP32-S2 some days ago. It shouldn't be a big deal.
Please let me know if you have a specific board selected in Arduino, I can test the linkage then.

**Comment by roguestreak** [comment]
Hi Marcel, thanks for the quick reply.
I'm a bit lost with all of the different versions of the ESP32 as to what their capabilities are, and what the pi pico's got; for example, the pi pico and ESP32-C3 both seem to mention USB, I've seen the pi pico being used to receive from a USB MIDI keyboard but not sure whether it would have enough spare capabilities to run the sound engine on top of this?  I've seen some excitement over the RP240's DMA, but haven't got the understanding of how much load that can take off the processor itself for running peripherals?
I guess it's a question of how much a single core is capable of, and whether the hardware interfaces of the new chips is able to take some of the load of the processor itself while giving USB without an expensive (comparatively) MAX3421E?

Unfortunately neither of my MIDI keyboards have MIDI-enabled knobs, and my "knobbed" keyboard is USB MIDI, so I'm sitting stuck between going for an external knob solution (which I've ordered the bits for) or a USB conversion / intermediary solution (though the USB MIDI keyboard only has 5 knobs)
